src/gluonts/model/deepar/_schema.py

```diff
@@ -0,0 +1,161 @@
+# permissions and limitations under the License.
+
+from dataclasses import Field
+from typing import Any, Dict, Optional, Union, Generic, TypeVar, Type
+
+import numpy as np
+import pandas as pd
+from functools import lru_cache
+from pandas.tseries.frequencies import to_offset
+from pandas.tseries.offsets import Tick
+from gluonts.exceptions import GluonTSDataError
+
+
+T = TypeVar("T")
+
+
+class FieldType(Generic[T]):
+    def __call__(self, v: Any) -> T:
+        raise NotImplementedError()
+
+    def __eq__(self, other):
+        raise NotImplementedError()
+
+    def is_compatible(self, v: Any) -> bool:
+        raise NotImplementedError()
+
+
+class NumpyArrayField(FieldType[np.ndarray]):
+    def __init__(
+        self,
+        dtype: Type = np.float32,
+        ndim: Optional[int] = None,
+    ) -> None:
+        self.dtype = dtype
+        self.ndim = ndim
+
+    def __eq__(self, other):
+        if not isinstance(other, NumpyArrayField):
+            return False
+        return self.dtype == other.dtype and self.ndim == other.ndim
+
+    def __repr__(self):
+        return f"NumpyArrayField(dtype={self.dtype!r}, ndim={self.ndim!r}"
+
+    def __call__(self, value: Any) -> np.ndarray:
+        value = np.asarray(value, dtype=object)
+        value = value.astype(self.dtype)
+        if self.ndim is not None and self.ndim != value.ndim:
+            raise GluonTSDataError(
+                f"expected array with dimension {self.ndim}, but got {value.ndim}."
+            )
+        return value
+
+    def is_compatible(self, value: Any) -> bool:
+        if not isinstance(value, (list, tuple, np.ndarray)):
+            return False
+
+        try:
+            x = np.asarray(value)
+        except:
+            return False
+
+        # int types
+        if self.dtype in [np.int32, np.int64]:
+            if x.dtype.kind != "i":
+                return False
+            return self.ndim is None or x.ndim == self.ndim
+
+        # float types
+        try:
+            x = np.asarray(value, dtype=self.dtype)
+        except:
+            return False
+        return self.ndim is None or x.ndim == self.ndim
+
+
+Freq = Union[str, pd.DateOffset]
+
+
+class PandasPeriodField(FieldType[pd.Period]):
+    def __init__(
+        self,
+        freq: Freq,
+    ) -> None:
+        self.freq = to_offset(freq) if isinstance(freq, str) else freq
+
+    def __eq__(self, other):
+        if not isinstance(other, PandasPeriodField):
+            return False
+        return self.freq == other.freq
+
+    def __repr__(self):
+        return f"PandasPeriodField(freq={self.freq!r})"
+
+    def __call__(self, value: Any) -> pd.Period:
+        period = PandasPeriodField._process(value, self.freq)
+        return period
+
+    @staticmethod
+    @lru_cache(maxsize=10000)
+    def _process(string: str, freq: pd.DateOffset) -> pd.Timestamp:
+        return pd.Period(string, freq=freq)
```

Comment:
I think we can just remove this. We used to do more complicated calculations here, but `pd.Period` is a lot faster than `pd.Timestamp`. So `__call__` should just be `return pd.Period(value, self.freq)`.